wlc/main.py

```diff
@@ -631,33 +632,111 @@ def run(self):
+            "-o",
+            "--output",
+            help="File|Directory where to store output (defaults to stdout)",
+        )
+        parser.add_argument(
+            "-g",
+            "--no-glossary",
+            action="store_true",
+            default=False,
+            help="Disable download glossary (defaults to False)",
         )
         return parser
 
+    def download_component(self, component):
+        """
+        Download a single component as file (if not a translation).
+        """
+
+        content = component.download(self.args.convert)
+        directory = Path(self.args.output)
+        file_path = directory.joinpath(f"{component.project.slug}-{component.slug}.zip")
+
+        if not directory.exists():
+            directory.mkdir(exist_ok=True, parents=True)
+
+        with open(file_path, "wb") as file:
+            file.write(content)
+
     def run(self):
         """Executor."""
-        obj = self.get_object()
-        content = obj.download(self.args.convert)
-        if self.args.output and self.args.output != "-":
-            with open(self.args.output, "wb") as handle:
-                handle.write(content)
-        else:
-            self.stdout.buffer.write(content)
+        if not self.args.object:
+            # All translations for all our projects
+            for component in self.wlc.list_components():
+                # Ignore glossary via --no-glossary
+                if component.slug == "glossary" and self.args.no_glossary:
+                    continue
+                self.download_component(component)
+                item = f"{component.project.slug}/{component.slug}"
+                self.println(f"downloaded translations for component: {item}")
+            return None
+
+        obj = self.wlc.get_object(self.args.object[0])
+
+        # Translation locale for a component
+        if isinstance(obj, wlc.Translation):
+            content = obj.download(self.args.convert)
+            if self.args.output and self.args.output != "-":
+                with open(self.args.output, "wb") as handle:
+                    handle.write(content)
+            else:
+                self.stdout.buffer.write(content)
+            return None
+
+        # All translations for a component
+        if isinstance(obj, wlc.Component):
+            for component in self.wlc.list_components():
+                # Only download for the component we scoped
+                if obj.slug == component.slug:
+                    self.download_component(component)
+                    self.println(
+                        f"downloaded translations for component: {self.args.object[0]}"
+                    )
+
+            return None
+
+        # All translations for a project
+        if isinstance(obj, wlc.Project):
+            for component in obj.list():
+                # Ignore glossary via --no-glossary
+                if component.slug == "glossary" and self.args.no_glossary:
```

Comment:
Better to look at the glossary attribute (I've exposed that just now in d6a52f0):

```suggestion
                if component.is_glossary and self.args.no_glossary:
```